### ISSUE TITLE
Add argument to open Oil in Preview-Mode

### ISSUE THREAD

**Issue by jdsee** [stevearc/oil.nvim#339]
### Did you check existing requests?

- [X] I have searched the existing issues

### Describe the feature

It would be great if `open` had an argument for enabling preview mode as soon as Oil opens.

Suggestion:
```lua
oil.open({
  dir = '...',
  preview = true,
})
```

### Provide background

I couldn't find a way to open Oil with preview-mode activated initially.

I tried the following but preview triggers too early this way:
```lua
require('oil').open()
require('oil.actions').preview()
``` 

### What is the significance of this feature?

nice to have

### Additional details

Edit: This is just an issue for opening Oil for the first time in a session. So maybe it would be better to offer an option for `setup` to enable preview by default.

**Comment by musicq** [comment]
It will be great If can support open in preview mode. Meanwhile, I'm using this trick to achieve it.

```
vim.keymap.set('n', '-', function()
  oil.open()

  vim.defer_fn(function()
    oil.select({ preview = true })
  end, 50)
end)

```

**Comment by SadmanTariq** [comment]
This is what I'm using as a workaround to try opening the preview only after its available

```lua
vim.keymap.set('n', '-', function()
  oil.open()

  -- Wait until oil has opened, for a maximum of 1 second.
  vim.wait(1000, function()
    return oil.get_cursor_entry() ~= nil
  end)
  if oil.get_cursor_entry() then
    oil.open_preview()
  end
end)
```

**Comment by stevearc** [comment]
This is now supported as part of the `oil.open` and `oil.open_float` API, or via the command with `:Oil --preview`
